WordPress/Classes/ViewRelated/Activity/ActivityUtils.swift

```diff
@@ -0,0 +1,95 @@
+import Foundation
+
+/// This is all based on Dennis Snell implementation on Calypso.
+/// Keeping the same function and parameters names to be able to re use the documentation.
+/// https://github.com/Automattic/wp-calypso/tree/master/client/state/activity-log/log/is-discarded
+///
+public class ActivityUtils {
+
+    /// Returns a memoized function.
+    ///
+    /// - Parameters:
+    ///     - body: The function to memoize.
+    ///
+    class func memoize<T: Hashable, U>(body: @escaping ((T)->U, T) -> U) -> (T)->U {
```

Comment:
I'd suggest separating this from ActivityUtils, and making it a bit more approachable. I was familiar with the concept of memoization but it still took a while to figure out what was going on here, because it wasn't clear it was designed to be recursive.

I have an alternate implementation with more descriptive type names here: https://gist.github.com/koke/7b9de2002b9666d3e9233c345bc51cb6

Reply:
Thanks for that, suggestion followed :)